src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdStateUpdateService.java

```diff
@@ -0,0 +1,113 @@
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus;
+import uk.gov.hmcts.reform.sscs.reference.data.mappings.CancellationReason;
+
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus.AWAITING_LISTING;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus.LISTED;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingStatus.FIXED;
+
+
+@Slf4j
+@Service
+public class CcdStateUpdateService {
+
+    public void updateListed(HearingGetResponse hearingResponse, SscsCaseData sscsCaseData) {
+        State state = State.UNKNOWN;
+        if (isHearingListingStatusFixed(hearingResponse)) {
+            state = mapHmcCreatedOrUpdatedToCcd(hearingResponse);
+        }
+
+        if (isKnownState(state)) {
+            sscsCaseData.setState(state);
+            log.info("CCD state has been updated to {} for caseId {}", state, sscsCaseData.getCcdCaseId());
+        }
+    }
+
+    public void updateCancelled(HearingGetResponse hearingResponse, SscsCaseData sscsCaseData) throws UpdateCaseException {
+        State state = State.UNKNOWN;
+
+        if (isHearingCancelled(hearingResponse)) {
+            state = mapHmcCancelledToCcdState(
+                hearingResponse.getHearingResponse().getHearingCancellationReason(), sscsCaseData.getCcdCaseId());
+        }
+        if (isKnownState(state)) {
+            sscsCaseData.setState(state);
+            log.info("CCD state has been updated to {} for caseId {}", state, sscsCaseData.getCcdCaseId());
+        }
+    }
+
+    public void updateFailed(SscsCaseData sscsCaseData) {
+        sscsCaseData.setState(State.HANDLING_ERROR);
+        log.info("CCD state has been updated to {} for caseId {}", State.HANDLING_ERROR, sscsCaseData.getCcdCaseId());
+    }
+
+    private State mapHmcCreatedOrUpdatedToCcd(HearingGetResponse hearingResponse) {
+        if (checkHearingStatus(hearingResponse, LISTED)) {
+            return State.HEARING;
+        }
+        if (checkHearingStatus(hearingResponse, AWAITING_LISTING)) {
+            return State.READY_TO_LIST;
+        }
+        return State.UNKNOWN;
+    }
+
+    private boolean isKnownState(State state) {
+        return !state.equals(State.UNKNOWN);
+    }
+
+    @SuppressWarnings("PMD.CyclomaticComplexity")
+    private State mapHmcCancelledToCcdState(String cancellationReason, String caseId) throws UpdateCaseException {
+        CancellationReason reason = CancellationReason.getCancellationReasonByValue(cancellationReason);
+
+        if (reason == null) {
+            UpdateCaseException exc = new UpdateCaseException(
+                String.format("Can not map cancellation reason with null value for caseId %s", caseId));
+            log.error(exc.getMessage(), exc);
+
+            throw exc;
+        }
```

Comment:
I think here we can avoid logging the exception.  Where it's caught will log it anyway, so this could simplify to just:

throw new UpdateCaseException(
                String.format("Can not map cancellation reason with null value for caseId %s", caseId));

Reply:
will do